cmds/core/time/time.go

```diff
@@ -33,35 +33,42 @@ package main
 import (
 	"flag"
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"os/exec"
 	"time"
 )
 
-func printTime(label string, t time.Duration) {
-	fmt.Fprintf(os.Stderr, "%s %.03f\n", label, t.Seconds())
+func printTime(stderr io.Writer, label string, t time.Duration) {
+	fmt.Fprintf(stderr, "%s %.03f\n", label, t.Seconds())
 }
 
-func main() {
-	flag.Parse()
-	a := flag.Args()
+func run(args []string, stdin io.Reader, stdout io.Writer, stderr io.Writer) error {
 	start := time.Now()
-	if len(a) == 0 {
-		fmt.Fprintf(os.Stderr, "real 0.000\nuser 0.000\nsys 0.000\n")
-		os.Exit(0)
+	if len(args) == 0 {
+		fmt.Fprintf(stderr, "real 0.000\nuser 0.000\nsys 0.000\n")
```

Comment:
suggestion:
printTime("real", time.Since(start))

This would ensure that printing is always consistent. You won't have this hard-coded string at line 50 that may not match what printTime might change to print at some future date. 
Further, print time could print a variadic
type labelTime struct {
name string
value int}

func printTime(w io.Writer, l ...labelTime)
Then you can call it
printTime(stderr, labelTime{name: "real", value: time.Since(..)}, labelTime{name: "user", time.c.ProcessState
etc. etc

the struct thing is not utterly necessary but it simplifies things a bit. 

I'm not sure this whole defer struct is needed. Could we not have user, and sys, two ints, that are zero; then run c.Run and, if it has no error, set user and sys, then call a print? 
I think that would be a bit simpler than this defer design.

Reply:
also, if you don't like the inline structs
fun label (n string, t time.Time) labelTime {
return labelTime(name: n, t: t}
or even

type labeler func(n string, t time.Time) string {
return fmt.Sprintf("%s:%v", n, t)
}

func printTime(w io.Writer, l ...labeler) {
for i := range l ...
}

I think printTime taking a variadic might simplify the code a tiny bit. Just what that variadic is, well, not sure.

Reply:
sigh, that's not quite right, but I hope it makes a small bit of sense. Sorry.

I suppose you could do this
func label(rn string, t time.Time) string {
return fmt.Sprintf("%s:%v", n, t)
}

for the print:
fmt.Fprintf("%s\n", strings.Join([]string{label("a", val), label("b", val),etc.}

just rambling. But I mainly don't like the hardcoded output, since it assumes a style that may change over time.

Reply:
Thank for review. Moved to format into one place.